### DISCUSSION TITLE
Is there a way to implement client-side caching ?

### DISCUSSION THREAD

**Discussion by BiitWanderer** [serilog-contrib/serilog-sinks-grafana-loki#169] [Q&A] [answered]
Hello.
I faced a problem that unsent logs wont reach the server if you restart the application (For example if batch size is small and amount of unsent logs in queue is large). I think that all unsent logs are stored in application's RAM. Could you tell me please if **there is a way to implement client-side caching so unsent logs would be stored on pc's drive until they sent ?** I did not find information about this in documentation and sink's configuration guide.

**Comment by mishamyte** [comment, marked as answer]
Hi @BiitWanderer,

For now there is no way for doing that, unfortunately. It was in old major sink version, but was cropped out, because it didn't worked with structured logging. It is in my mind for the future, but I doubt it would be in the nearest version.

So for now I could propose you to write in stdout, using [LokiJsonTextFormatter](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/blob/master/src/Serilog.Sinks.Grafana.Loki/LokiJsonTextFormatter.cs) and deliver it, using DevOps tools like `promtail` (so you could combine persistent mode and structured logging probably)

**Reply by BiitWanderer** [reply]
Thank you for quick and helpful response.
